### DISCUSSION TITLE
SOLVED! How to fix bug bootstrap modal backdrop in livewire 3 because wire:navigate

### DISCUSSION THREAD

**Discussion by ilahazs** [livewire/livewire#7533] [7. Bugs (archived)]
### Livewire version

v3.2.6

### Laravel version

v10.0.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

Weird bug.
![image](https://github.com/livewire/livewire/assets/68785569/fe9626e5-31d6-4ad2-9adb-d825343cc0cd)

Steps to reproduce.
1. Use Bootstrap 5
2. Launch the bootstrap 5 Modal
3. Navigate to other page using wire:navigate
4. Comes back to page that has modal
5. Launch the modal again
6. You will see the backdrop seems darker than usual (It's because the backdrop is stacked)
7. Put the script below

`$.fn.modal.Constructor.prototype.show = function () {
   originalModalShow.apply(this, arguments);

   var modalBackdrops = $('.modal-backdrop.fade.show');
   for (var i = 1; i < modalBackdrops.length; i++) {
      $(modalBackdrops[i]).remove();
   }
};`

9. And try it again

### Repository or Wirebox URL

github.com/livewire/livewire

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by Higor12cs** [comment]
Did you solve it? I'm having the same problem.

**Comment by eqbilhadi** [comment]
I have tried it but it still doesn't work

**Reply by ahmadrivaldi-arv** [reply]
I've just resolved this issue. Simply add this to your script tag.

#### jQuery
```js
$.fn.modal.Constructor.prototype.show = () => $('.modal-backdrop').not(":first").remove()
```

put that in your layout

**Comment by ahmadrivaldi-arv** [comment]
this problem appear while using wire:navigate. The more frequent the navigation, the darker the background becomes.

this code solved the problem

> I've just resolved this issue. Simply add this to your script tag.
> 
> #### jQuery
> ```js
> $.fn.modal.Constructor.prototype.show = () => $('.modal-backdrop').not(":first").remove()
> ```
> 
> put that in your layout

**Reply by Yapidev** [reply]
Thanks bro, it's works for me hahaha

**Reply by kriit24** [reply]
it does not works for me, now modal does not work at all

**Comment by sagadsalem** [comment]
Any Update about this issue?

**Comment by ahid-technologies** [comment]
Still remains mystery? 🥴
I am using livewire navigation and instances of modals from previous pages are being carried forward to subsequent pages, resulting in multiple backdrops being rendered.

**Comment by firer777** [comment]
add document.addEventListener('livewire:navigated', () => { // remove backdrop not use });

**Reply by ahid-technologies** [reply]
I tried it but it did not worked, I had to move my theme JS to @assets directive of the livewire. Not the best fix, but it is working. 👀

**Reply by firer777** [reply]
@ahid-technologies u can u event livewire:navigating on js or use @livewire:navigating.window="..." on div contain modal for destroy, close or remove backdrop before navigated

**Reply by firer777** [reply]
example use navigating event close modal before page navigated

**Reply by firer777** [reply]
i using livewire bootstrap modal package https://github.com/wire-elements/modal

**Comment by ahid-technologies** [comment]
⚠Attention guys ⚠
I HAVE FINALLY FOUND THE SOLUTION.

You just need to add this attribute to your bootstrap script tag "data-navigate-once", It will stop livewire from re-evaluating the script each time it navigates. Effectively solving the problem.
`<script src="{{ asset('libs/bootstrap/js/bootstrap.bundle.min.js') }}" data-navigate-once></script>`

Refer to the docs here for more information: 
https://livewire.laravel.com/docs/navigate#reloading-when-assets-change
